extension/parquet/parquet_extension.cpp

```diff
@@ -185,6 +185,21 @@ static MultiFileReaderBindData BindSchema(ClientContext &context, vector<Logical
 	return_types = schema_col_types;
 	D_ASSERT(names.size() == return_types.size());
 
+	if (options.file_row_number) {
+		if (std::find(names.begin(), names.end(), "file_row_number") != names.end()) {
+			throw BinderException(
+			    "Using file_row_number option on file with column named file_row_number is not supported");
+		}
+
+		bind_data.file_row_number_idx = names.size();
+		return_types.emplace_back(LogicalType::BIGINT);
+		names.emplace_back("file_row_number");
```

Comment:
This code is now duplicated between `parquet_extension.cpp` and `parquet_reader.cpp`. Would it be possible to unify?

Reply:
We discussed this, and the code would not be more readable if we de-duplicate it

Reply:
Let me add the reasoning here:

With a provided schema in the BindSchema, we are binding based on the provided schema and the parquet options. Then later when the parquet reader is initialized, the parquet reader needs to initialize itself with the file_row_number column (which needs to check the option again.)

Without a provided schema we are binding based on what we read in the ParquetReader, where the reader itself is initialized with the file row number option, then the bind is done on all columns produced by the reader.

Because the hypothetical function to deduplicate this, maybe called `BindParquetOptions()` or `AddParquetOptions()`, would be semantically pretty confusing, i think it would create more problems than it solves.